### PR TITLE
executor: fix 'set config' string escaping

### LINKED ISSUE #31258: executor: wrong JSON format when convert string config item
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Modify configuration online and check out the warning message:
```sql
set config tikv `rocksdb.defaultcf.compression-per-level`= '["no","no","lz4","lz4","lz4","zstd","zstd"]';

show warnings;
```
### 2. What did you expect to see? (Required)

No warning message.

### 3. What did you see instead (Required)

```
  Level  | Code |                                                         Message                                                          
---------+------+--------------------------------------------------------------------------------------------------------------------------
 Warning | 1105 | bad request to http://127.0.0.1:52751/config: failed to decode, error: Error("expected `,` or `}`", line: 1, column: 47) 
(1 row)
```

### 4. What is your TiDB version? (Required)

```
Release Version: v5.3.0
Edition: Community
Git Commit Hash: 4a1b2e9fe5b5afb1068c56de47adb07098d768d6
Git Branch: heads/refs/tags/v5.3.0
UTC Build Time: 2021-11-24 13:31:24
GoVersion: go1.16.5
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #31258

Problem Summary:

### What is changed and how it works?

``"%s"`` doesn't handle the escape correctly when there are `"` in the input string, change to `%q`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/set_config.go [hunk 1]
```diff
@@ -120,6 +120,7 @@ func (s *SetConfigExec) Next(ctx context.Context, req *chunk.Chunk) error {
 
 func (s *SetConfigExec) doRequest(url string) (retErr error) {
 	body := bytes.NewBufferString(s.jsonBody)
+	fmt.Println("do request json body ==", s.jsonBody)
```
Comment (Defined2014) [inline executor/set_config.go:123]: delete it?

Comment (tiancaiamao) [reply executor/set_config.go]: Done

**Review by Defined2014 — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**